modules/gpgsqlbackend/gpgsqlbackend.cc

```diff
@@ -52,6 +52,8 @@ gPgSQLBackend::gPgSQLBackend(const string &mode, const string &suffix)  : GSQLBa
     g_log<<Logger::Error<<mode<<" Connection failed: "<<e.txtReason()<<endl;
     throw PDNSException("Unable to launch "+mode+" connection: "+e.txtReason());
   }
+
+  reconnect();
```

Comment:
This seems a bit wasteful since it will close the connection and establish a new one, while we likely just did that? I'm guessing there is a reason you are not calling `allocateStatements()` here but it's not obvious to me.

Reply:
You're quite right, calling `reconnect()` here is wrong. I've pushed a change to use `allocateStatements()` here but will verify this works on my test branch!